background/providers/torrentproject.js

```diff
@@ -0,0 +1,99 @@
+var https = require('https');
+var zlib = require('zlib');
+var path = require('path');
+var nconf = require('nconf');
+var mongoose = require('mongoose');
+var Category = require('../../app/models/Category.js');
+var Torrent = require('../../app/models/Torrent.js');
+
+nconf.use('memory');
+nconf.argv().env('__').file({
+    file: path.resolve(__dirname + '/../../config.json')
+});
+
+if(nconf.get('database:mongodb:enabled')){
+    mongoose.connect('mongodb://' + nconf.get('database:mongodb:host') + ':' + nconf.get('database:mongodb:port') + '/' + nconf.get('database:mongodb:collection'), function(err){
```

Comment:
What's this supposed to do? On my computer (Windows 10, Node v5.5.0 mongoose connects fine but `err` still exists with the following error:

```JavaScript
{ [Error: Trying to open unclosed connection.] state: 2 }
```

Can we ignore this error since the database connection is already open or should we be using `createConnection` instead of `connect`?